src/components/modal/modalCo.js

```diff
@@ -0,0 +1,320 @@
+  },
+  /**
+   * Variables to make the CloseSVG button stay in place
+   */
+  CloseSVG: {
+    float: "right",
+    position: "sticky",
+    top: "5vh",
+    width: 0,
+  },
+}
+
+/**
+ * Creates an stylized Action button to apply the dynamic changes
+ */
+const ActionButton = withStyles(styles)(({ classes, type, ...other }) => {
+  return <ButtonCo classes={{ root: classes.btnRoot }} {...other} />
+})
+
+/**
+ * Creates an stylized Action adrea to apply the dynamic changes
+ */
+const ActionsArea = withStyles(styles)(({ classes, type, ...other }) => {
+  return (
+    <DialogActions
+      classes={{ root: classes.actions, spacing: classes.actionsSpacing }}
+      {...other}
+    />
+  )
+})
+/**
+ * Creates an stylized paper for the dialog
+ */
+const Modal = withStyles(styles)(({ classes, ...other }) => {
+  return (
+    <Dialog
+      classes={{
+        paper: classes.paper,
+      }}
+      {...other}
+    />
+  )
+})
+/**
+ * Sets the definition for the CloseButton
+ */
+const CloseButton = withStyles(styles)(({ classes, ...other }) => {
+  return (
+    <IconButton
+      color="primary"
+      size="small"
+      {...other}
+      classes={{
+        root: classes.CloseSVG,
+      }}
+    >
+      <CloseIcon style={{ fontSize: 52 }} />
+    </IconButton>
+  )
+})
+
+/**
+ * Function that sets the primary/unique button for the Actions
+ *
+ * @param {string} text Text to be placed as the label for the button
+ * @param {function} action Action to be performed when the button is clicked
+ */
+const justPrimary = (text, action) => {
+  return (
+    <ActionButton type="primary" fullWidth onClick={action} end={Right}>
+      <div style={styles.btnLabel}>{text}</div>
+    </ActionButton>
+  )
+}
+/**
+ * Function that sets both the secondary and primary button for the Actions
+ *
+ * @param {[string, function]} primary Text and Action to be performed by the primary button
+ * @param {[string, function]} secondary Text and Action to be performed by the secondary button
+ */
+const primaryAndSecondary = (primary, secondary) => {
+  return [
+    <ActionButton
+      key="secondary"
+      type="secondary"
+      color="primary"
+      variant="outlined"
+      onClick={secondary[1]}
+      fullWidth
+    >
+      <div style={styles.btnLabel}>{secondary[0]}</div>
+    </ActionButton>,
+    <ActionButton key="primary" type="primary" fullWidth onClick={primary[1]}>
+      <div style={styles.btnLabel}>{primary[0]}</div>
+    </ActionButton>,
+  ]
+}
+
+/**
+ * Default props to set for the Modal
+ */
+const defaultProps = {
+  fullWidth: true,
+  maxWidth: "lg",
+}
+
+function ModalCo(props) {
+  const {
+    children,
+    primarybutton,
+    secondarybutton,
+    showClose,
+    onClose,
+    onPrimary,
+    onSecondary,
+  } = props
+  const currentProps = { ...defaultProps, ...props }
+
+  /**
+   * Sets the action portion of the modal, according to the props
+   */
+  let modalButton
+  if (primarybutton !== "") {
```

Comment:
what do you think about using the following statement.. 

````
 
if(!primarybutton){
........
````

The above case you validate empty and undefined values

Reply:
Sometimes I forget about that, thanks!